src/buildercore/terraform.py

```diff
@@ -362,7 +359,7 @@ def _generate_vcl_file(stackname, content, key, extension='vcl'):
     """
     with _open(stackname, key, extension=extension, mode='w') as fp:
         fp.write(str(content))
-        return '${file("%s")}' % basename(fp.name)
+        return '${file("%s")}' % os.path.basename(fp.name)
```

Comment:
`from os.path import join` is inconsistent with `os.path.basename()`, which is the chosen style?

Reply:
personally I use more `os.path.*(...)` calls as they declare their dependency

Reply:
> from os.path import join is inconsistent with os.path.basename(), which is the chosen style?

yeah, it's an exception to the rule. It *is* consistent with all my other code though. And I've not seen `from os.path import basename` anywhere else (yet) at elife. I don't really care though, I change code as I read through it and this was just one case I couldn't be bothered reverting before commit. I'll leave it be from now on if I see it happening elsewhere.

Reply:
In general I would prefer `from ... import ...` but _join_(`from os.path import join` ) is too generic. `pathlib` can help with that and we should use it more often I think.

Reply:
"too generic" - I disagree. There is no builtin called "join" and I've never written a function called "join" nor kicked myself for shadowing the imported "join" with another's "join" function.

I think this is getting out of hand. Beyond the stuff controlled by `autopep8` can we agree that different authors of code have different styles? and that coercing others to conform to your own personal style so it's easier for *you* to read, as if you were some sort of authority on universal code readability, is a jerky thing to do? I try not to pick at other's style beyond the obvious bad patterns, but this time I trod on @giorgiosironi toes.

Reply:
... the emphatic 'you' wasn't any specific individual.

Reply:
That's why we use `autopep8` and other coding standard tools, so that there is an arbitrary choice made by it and people don't have to discuss it :) so if it's not enforced there, there is no need to edit war - I see this mostly as knowledge sharing as I have become aware of `pathlib` for example which I didn't know before